# TAREA FINAL - Yarim Kamila Gamarra Dávila
## I. Pandas

### 1.

In [1]:
!pip install pandas

In [2]:
import pandas as pd

In [3]:
data_ciudades = {
    'Ciudad': ['Lima', 'Arequipa', 'Trujillo', 'Cusco', 'Chiclayo', 'Piura', 'Iquitos', 'Huancayo', 'Tacna', 'Pucallpa'],
    'Habitantes': [1047996, 100169, 92331, 428450, 305717, 484475, 441649, 385098, 294395, 283734],
    'Area_km2': [2672.28, 1545.77, 1487.7, 116.5, 279.89, 6217.26, 368.9, 109.19, 59.4, 483.44],
    'Altitud_m': [154, 2325, 34, 3399, 29, 29, 106, 3271, 562, 156],
    'Densidad_poblacion_por_km2': [3924, 64.8, 62.1, 3673, 1091.6, 77.9, 1196, 3528.6, 4966, 587.2]    
}


In [9]:
info=pd.DataFrame(data_ciudades)
info

Ciudad  Habitantes  Area_km2  Altitud_m  Densidad_poblacion_por_km2
0      Lima     1047996   2672.28        154                      3924.0
1  Arequipa      100169   1545.77       2325                        64.8
2  Trujillo       92331   1487.70         34                        62.1
3     Cusco      428450    116.50       3399                      3673.0
4  Chiclayo      305717    279.89         29                      1091.6
5     Piura      484475   6217.26         29                        77.9
6   Iquitos      441649    368.90        106                      1196.0
7  Huancayo      385098    109.19       3271                      3528.6
8     Tacna      294395     59.40        562                      4966.0
9  Pucallpa      283734    483.44        156                       587.2

### 2.

In [10]:
data_ciudades_2 = {
    'Ciudad': ['Lima', 'Arequipa', 'Trujillo', 'Cusco', 'Chiclayo', 'Piura', 'Iquitos', 'Huancayo', 'Tacna', 'Pucallpa'],
    'Gentilicio': ['Limense', 'Arequipeño', 'Trujillano', 'Cusqueño', 'Chiclayano', 'Piurano', 'Iquiteño', 'Huancaino', 'Tacneño', 'Pucallpino'],
    'Provincia': ['Lima', 'Arequipa', 'Trujillo', 'Cusco', 'Chiclayo', 'Piura', 'Maynas', 'Huancayo', 'Tacna', 'Coronel Portillo'],
    'Region': ['Lima', 'Arequipa', 'La Libertad', 'Cusco', 'Lambayeque', 'Piura', 'Loreto', 'Junín', 'Tacna', 'Ucayali']
}


In [11]:
nombres=pd.DataFrame(data_ciudades_2)
nombres

Ciudad  Gentilicio         Provincia       Region
0      Lima     Limense              Lima         Lima
1  Arequipa  Arequipeño          Arequipa     Arequipa
2  Trujillo  Trujillano          Trujillo  La Libertad
3     Cusco    Cusqueño             Cusco        Cusco
4  Chiclayo  Chiclayano          Chiclayo   Lambayeque
5     Piura     Piurano             Piura        Piura
6   Iquitos    Iquiteño            Maynas       Loreto
7  Huancayo   Huancaino          Huancayo        Junín
8     Tacna     Tacneño             Tacna        Tacna
9  Pucallpa  Pucallpino  Coronel Portillo      Ucayali

### 3.

In [15]:
cuadro_1 = pd.merge(info, nombres, how='inner')
cuadro_1

Ciudad  Habitantes  Area_km2  Altitud_m  Densidad_poblacion_por_km2  \
0      Lima     1047996   2672.28        154                      3924.0   
1  Arequipa      100169   1545.77       2325                        64.8   
2  Trujillo       92331   1487.70         34                        62.1   
3     Cusco      428450    116.50       3399                      3673.0   
4  Chiclayo      305717    279.89         29                      1091.6   
5     Piura      484475   6217.26         29                        77.9   
6   Iquitos      441649    368.90        106                      1196.0   
7  Huancayo      385098    109.19       3271                      3528.6   
8     Tacna      294395     59.40        562                      4966.0   
9  Pucallpa      283734    483.44        156                       587.2   

   Gentilicio         Provincia       Region  
0     Limense              Lima         Lima  
1  Arequipeño          Arequipa     Arequipa  
2  Trujillano          Trujillo  La Libertad  
3    Cusqueño             Cusco        Cusco  
4  Chiclayano          Chiclayo   Lambayeque  
5     Piurano             Piura        Piura  
6    Iquiteño            Maynas       Loreto  
7   Huancaino          Huancayo        Junín  
8     Tacneño             Tacna        Tacna  
9  Pucallpino  Coronel Portillo      Ucayali

### 4.

In [17]:
cuadro_1.describe()

Habitantes    Area_km2    Altitud_m  Densidad_poblacion_por_km2
count  1.000000e+01    10.00000    10.000000                   10.000000
mean   3.864014e+05  1334.03300  1006.500000                 1917.120000
std    2.679283e+05  1917.13852  1410.508754                 1892.333136
min    9.233100e+04    59.40000    29.000000                   62.100000
25%    2.863992e+05   157.34750    52.000000                  205.225000
50%    3.454075e+05   426.17000   155.000000                 1143.800000
75%    4.383492e+05  1531.25250  1884.250000                 3636.900000
max    1.047996e+06  6217.26000  3399.000000                 4966.000000

La mínima densidad poblacional es 62.100000, correspondiente a la ciudad de Trujillo y la máxima es de 4966.000000, correspondiente a la ciudad de Tacna.

## II. Control de Flujo

### 1.

In [54]:
X = 6

if x%2==0 and x%3==0:
    print("Es múltiplo de 2 y 3")
elif x%2==0:
    print("Es múltiplo de 2")
elif x%3==0:
    print("Es múltiplo de 3")

else:
    print("Otro caso")

Es múltiplo de 2 y 3


### 2.

In [56]:
for x in range (1,31) :
    
    if x%3==0 and x%5==0:
        print("FizzBuzz")
    elif x % 3==0:
        print("Fizz")
    elif x % 5==0:
        print("Buzz")

    else:
        print(x)  


1
2
Fizz
4
Buzz
Fizz
7
8
Fizz
Buzz
11
Fizz
13
14
FizzBuzz
16
17
Fizz
19
Buzz
Fizz
22
23
Fizz
Buzz
26
Fizz
28
29
FizzBuzz


## III. Funciones

In [67]:
def calcular_craest(media_personal, media_curso, desviacion_curso):
    '''
    Objetivo:
        - Calcular el craest de un estudiante dado su media personal, la media del curso y la desviación estándar del curso
    
    Input:
        - media_personal: promedio del 0-20 del estuadiante
        - media_curso: promedio del  0-20 del salón
        - desviacion_curso: dispersión de notas del salón

    Output:
        - craest calculado
    '''
    craest = (((media_personal - media_curso)* 10)/desviacion_curso) + 50
    return craest

In [68]:
resultado=calcular_craest(14,15,3.5)
print(resultado)

47.142857142857146
